src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionBasicTests.cs

```diff
@@ -89,6 +89,21 @@ public void ClosedConnectionSchemaRetrieval()
             }
         }
 
+        [Theory]
+        [InlineData("RandomStringForTargetServer", true)]
```

Comment:
Is there a test that checks the equivalent of this but with false?

Reply:
I was too focused on testing what was reported in the bug. I will add a test case which checks that the Connection.WorkstationId == "RandomStringForTargetServer" in a non-disposed connection.
Thanks